modules/contact/src/actions/ContactAction.C

```diff
@@ -108,8 +108,16 @@ ContactAction::validParams()
       "normal_lm_scaling", 1., "Scaling factor to apply to the normal LM variable");
   params.addParam<Real>(
       "tangential_lm_scaling", 1., "Scaling factor to apply to the tangential LM variable");
-  params.addParam<bool>("use_dual", false, "Whether to use the dual mortar approach");
```

Comment:
Great! Thanks for keeping the option here!

Reply:
The addition of standard bases (`use_dual = false`) with the weighed approach assumes same-order bases, which is just one way to do it, since all combinations have shown some result instabilities (including mixed-order bases).

I'd envision that once/if we remove the `legacy` option from the action, we may not have to stress whether we are using dual bases or weighted-quantity constraints, so it may be less confusing then.